packages/@aws-cdk/aws-ecs/lib/ec2/ec2-service.ts

```diff
@@ -198,6 +199,50 @@ export class Ec2Service extends BaseService implements elb.ILoadBalancerTarget {
     });
   }
 
+  /**
+   * Enable CloudMap service discovery for the service
+   */
+  public enableServiceDiscovery(props: ServiceDiscoveryProps): cloudmap.Service {
+    // Determine DNS type based on network mode
+    const networkMode = this.taskDefinition.networkMode;
+
+    if (networkMode === NetworkMode.None) {
+      throw new Error("Cannot use a service discovery if NetworkMode is None. Use Host or AwsVpc instead.");
+    }
+    // Bridge or host network mode requires SRV records
+    let dnsType = props.dnsType;
+
+    if (networkMode === NetworkMode.Bridge || networkMode === NetworkMode.Host) {
+      if (dnsType !== cloudmap.DnsType.Srv) {
+        throw new Error("Cannot use a service discovery if NetworkMode is None. Use Host or AwsVpc instead.");
+      }
+      dnsType = cloudmap.DnsType.Srv;
+    }
+
+    if (networkMode === NetworkMode.AwsVpc && props.dnsType === undefined ) {
+      // ECS does not currently support IPv6, so AAAA records not possible
+      dnsType = cloudmap.DnsType.A;
+    }
+
+    const cloudmapService = new cloudmap.Service(this, 'CloudmapService', {
+      namespaceId: props.namespaceId, // FIXME use INamespace instead -- validate if cluster has called addNamespace set?
```

Comment:
Why are you not taking the `namespaceId` from the cluster in any case? It's there, why make the user pass it in again? And in case there can be multiple namespaces why store it on the cluster at all?

Why *does* `cluster.addNamespace()` exist, anyway?